### ISSUE TITLE
When is the Chromium version up?

### ISSUE THREAD

**Issue by moon1618** [kiwibrowser/src#94]
Even if you look at All workflows, the version of Chromium that succeeded in building recently isIt is 77.03865.116.
Too old
The current Chromium is 81.0.4044.138.
Are you waiting for others to upgrade due to being open source?
Will master upgrade?
Will Kiwi end as it is?
I'm very anxious.
Please give an accurate answer.

**Comment by kiwibrowser** [comment]
Hello,

What feature from 81 do you miss ?
For now the recent versions of Chromium are significantly slower and more unstable on my test devices but if there is advantages to follow Chromium on this why not but the goal is to have something different from Google

**Comment by joergRossdeutscher** [comment]
We are missing all security fixes since last year.

**Comment by Akianonymus** [comment]
Well, i don't know what got missed, but this article covers the latest rollout

https://www.androidpolice.com/2020/05/20/chrome-83-introduces-new-privacy-features-revamps-form-controls-and-more-apk-download/

Also, i was wondering how much "significantly slow and more unstable" it was ? Did you benchmark it ? Does that affect in real life usage ? Just some questions

I get how this project aims to be different from google, but doesn't mean we have to skip things, atleast important features and security patches.

Again, i am not a developer so I don't know the difficulty of what i have said above, just sharing my thoughts.

**Comment by moon1618** [comment]
Hello, Owner

It says it's significantly slower and unstable, does that mean Ibuilt a new Chromium with kiwi and verified?

If so, I too want to check the.
Please provide it in apk even in this comment section.

I would like to actually confirm how unstable and slow it is.

Other browsers have unstable Dev, Canary, etc., which can be confirmed.
It should be published on Github even if it is slow and unstable.

After confirming that, everyone should think about the correction method.

Anyway, security and bug updates are important.

I'm not good at English so it's hard to understand, but I'm sorry.

**Comment by joergRossdeutscher** [comment]
I don't demand any feature from a Free Software project. That said, there is one thing users definitely have to know:

Is this a "serious" project, targeting a daily driver app? Then quickly applying security fixes is totally mandatory in the future. Always. Reliable. Fast. This is a browser. It's a highly vulnerable software.

Or is this project some "hack fun" POC? Then have fun, but please do not encourage people to seriously use that app, because you put them in danger of being exploited.

Telling users current builds would go directly to the play store when that version contains every known exploit since october 2019 is just not fair.

**Comment by rururux** [comment]
now this is open source, you can send PR...

**Comment by joergRossdeutscher** [comment]
What has a PR to do with sharing insecure software? The whole process and discussions here make it clear that security is currently not a topic. It's not other people's job to fix this. 

At least the app should be removed from the PlayStore until someone is caring seriously, and the repo should say clearly “Do not use this, it has known security issues.“

**Comment by bitsper2nd** [comment]
Are a you security expert? do you have any substantial proof that current kiwi browser can be exploited? If your are not contributing to the project or have any actual claims to a flaw, your word means nothing. Want a more up to date chromium version, then use google chrome, brave or bromite.

**Comment by moon1618** [comment]
because of the potential for abuse
The new chromium version has several security fixes.

Isn't it safer to use a new chromium version to eliminate such risks beforehand?

Isn't it a different problem to use google chrome, brave, and bromine? These do not support extensions.

We're talking about Kiwi with extensions.

**Comment by SuperSandro2000** [comment]
If you are using older Chromium versions you are basically vurnable to all patched CVEs and for many of them PoC exist which make exploiting them really easy. If we would use a custom browser agent this would be easily exploitable. As long as you only browse safe sites with no user content or advertisements and those sites get not hacked you should be somewhat safe.

**Comment by Snaxeko** [comment]
> Are a you security expert? do you have any substantial proof that current kiwi browser can be exploited? If your are not contributing to the project or have any actual claims to a flaw, your word means nothing. Want a more up to date chromium version, then use google chrome, brave or bromite.

You don't need to be a security expert, just look at the chromium updates logs you'll get why Kiwi needs an update. 

[Here](https://chromereleases.googleblog.com/2020/04/stable-channel-update-for-desktop_7.html?m=1)  is an example of recent security patches, you can click on each security threats if you need details.

**Comment by SuperSandro2000** [comment]
I am not sure how much of the vulnerabilities from Chromium Desktop apply to Chromium mobile but some statistics to make this worse.

- The Debian Package listed 190 CVEs fixed since version 78.
- Google awarded 4 CVEs with over $10K or more in the last 5 weeks.

**Comment by Snaxeko** [comment]
> Are a you security expert? do you have any substantial proof that current kiwi browser can be exploited? If your are not contributing to the project or have any actual claims to a flaw, your word means nothing. Want a more up to date chromium version, then use google chrome, brave or bromite.

You don't need to be a security expert, just look at the chromium updates logs you'll get why Kiwi needs an update. 

[Here](https://chromereleases.googleblog.com/2020/02/chrome-for-android-update.html?m=1) is a little and comprehensive example of an improvement. But it's nothing compared to the big amount of upgrades since 2019.
The [last update](https://chromereleases.googleblog.com/2020/05/chrome-for-android-update_19.html?m=1) is a huge security update.

**Comment by kiwibrowser** [comment]
@joergRossdeutscher Which major CVE do you see unfixed that is not in Kiwi yet or is a concern to you ?
The RCE are not reproducible as far as I know.

The new versions of Chromium also adds new bugs, it's not just one way.

For now Kiwi it's basically the same engine as in Samsung Browser for example, or Opera or others.

It's public so people can audit and suggest changes.

Really, if you see a bug (of any nature, security, not security) if you see it fixed in Chromium and want Kiwi to include it, you can show it here and as soon as the lines of code are imported you'll get an APK automatically with the fixes.

Let's work productively then, do you have a commit or a bug that worries you that we should look at ?

With the regards of the Play Store builds, they'll get in sync with GitHub in short amount of time (to automate as much the pipeline as possible)

**Comment by SuperSandro2000** [comment]
> The RCE are not reproducible as far as I know.

There is a potential RCE which is not patched.

**Comment by kiwibrowser** [comment]
Here's the plan:

- Accept some fixes (and/or back port some of them)
It's essentially maintenance mode. No big new features, mostly cleaning up bugs.

The goal is to decrease number of bugs (of any kind), and improving the stability / security only.

Really, if you have a fix (security, or anything) you want to see, please point to the specific one, there are tons of bugs (security, not security, etc) in Chromium, not all applies to Kiwi, not all require hotfix.

Regarding bigger advances, the code of Kiwi has been taken by ungoogled-chromium, Brave, another big commercial browser, etc. So it will continue to live in its own way.

If there is an opportunity to sync then, of course but it will take a couple of months realistically.

**Comment by bitsper2nd** [comment]
You people blow everything up from proportion. Chances are that an user in chrome is more vulnerable than a kiwi user. The reason people use kiwi is for extensions, not security or privacy. That said, this is a niche browser for small audience developed by one person. Security fixes and bug fixes will come at a lower pace. If you don't like that, then use another browser. You are all making an assumption from reading a chromium change log. You all have never done a proper testing to determine if it really affects kiwi. You could be right, but there is no process behind the conclusion. Common security practice is to find a flaw a show it, so it can be reproduce and then fixed. I still plan on using kiwi. Not being up to date with the latest number version is a no problem for me. I know how deal with it first-hand. I have common sense. Since Kiwi supports extensions, I use ublock origin on medium.

**Comment by joergRossdeutscher** [comment]
As I said, I do not demand someone works for me. But, with every respect, this discussion sounds like 20 years ago. Not following upstream, asking for users to “prove“ exploits, declaring security as “not the main goal of this software“ is not best practice in 2020. 

I will not disturb this project by starting meta-discussions, but I will move on to a different browser. 

Thanks for the work — having addons in a mobile browser really, really rocks.

**Comment by kiwibrowser** [comment]
I get what you mean, but you paraphrase a bit much (I do understand why as well).

It's your own conclusions, I never said what you quoted (it's always more subtle as it looks, but I think it's a shortcut from all perspectives).

Though, thanks for the kind words too, and again, your opinion is not disturbing, and rather welcome, this is one of the choice of having the development public.

With regards of being different, it's a good thing.

For now with Kiwi, I want to postpone as much as possible the resyncing with upstream Chromium.

Yes it's easier just to follow Google, but here there are some fundamental product design disagreements that are worth it (e.g. Manifest V2); though, they may not be true in 1 year.

In the long-term, Kiwi will likely resync with Chromium.

I'm doing my best to align with Google and other browser makers to include Kiwi changes upstream (this will take a couple of months, and is in progress, but I'm like 90% sure that you will see other Chromium-Kiwi-based browsers very soon).

This way, maintainability is split among different actors (it's a big work).

I want to shift the burden of maintainability to them, it's not a secret, it's one of the millions of reason of open-sourcing (plus the fact that it's really good for the developers in general).

So this is the answer to the main topic (When is the version of Chromium changed).

Regarding fixes and bugs:

If there is a fix that someone wants to be included in Kiwi, the PR button is one-click away.
I'm always here to review code.

Otherwise, I'll take the important CVEs (and bugfixes in general) that are affecting Chromium, determine whether they affect Kiwi, if it does, then I implant the fix.

I said be patient, it will take time, there is one guy doing it's me.

At the end of the whole cycle, we should have a pretty much automated pipeline (from Chromium upstream to Kiwi release).

So Kiwi will essentially become a Chromium that has been freed/unlocked from restrictions.

Do you see and understand the vision and/or agree with the strategy ?

Let me know,

**Comment by kiwibrowser** [comment]
Updated Roadmap @ README.md to clarify this point

**Comment by kiwibrowser** [comment]
FYI checked the CVE-2020-6457 that most users are are worried about, Kiwi not affected

https://chromium.googlesource.com/chromium/src/+/refs/tags/81.0.4044.112..81.0.4044.113

It's in speech recognizer, a component that Kiwi doesn't use. It affects actually only Chrome-official builds.

**Comment by kiwibrowser** [comment]
Checking others

**Comment by kiwibrowser** [comment]
Ok, so, agreement is the following:

A) Do a couple of small fixes for now

B) It's not clear what CVEs exist in Chromium (Google keeps the technical details secret unfortunately), so instead of the strategy is to rebase automatically on a recent version of Chromium.

This will take some time, but here is the plan.

Will that resolve the issue and answer the question ?

**Comment by BeingDatta** [comment]
dear developer its true man u need to update chromium base soon very soon one of the most rich feature is not available in your version and that is U2F SECURITY KEY in chrome 81 its available in firefox its available trust me its important.

**Comment by bitsper2nd** [comment]
That is not really important for kiwi. If you are in mobile use a password manager app or any other app(most popular social media and cloud services) that supports U2F instead.

**Comment by moon1618** [comment]
Hello, Owner

Thank you for your specific explanation.
I understood your idea.
I look forward to future plans.

This will close it. 
Thank you.

**Comment by kiwibrowser** [comment]
Planned, work will start on https://github.com/kiwibrowser/src.next

**Comment by kiwibrowser** [comment]
A little heads up here, rebasing is still in the pipes, I'm waiting on browsers to take-over ownership on the extensions code and will move forward with this.

**Comment by bertandzobrist** [comment]
@kiwibrowser this work hasnt started yet or its already abandoned like any other foss developement by kiwi really shame on you guys just creating new repo to showoffs.

Lots of features lacking on previous kiwi which hasnt been updated for nearly one year and whenever feedback is provided the reply something like this that ........

1. this feature is not necessary
2. presently kiwi is more stable than chrome in reality which is a lameful shameful scam.
3. lastly most of the improvement suggestions are redirected here where there is no symbol of work done.

**Comment by bertandzobrist** [comment]
pls update the chrome version or atleast admit that you guys cant do that.

**Comment by bertandzobrist** [comment]
@kiwibrowser no work is there only readme and license

**Comment by bertandzobrist** [comment]
@kiwibrowser  no work is there only readm

**Comment by bitsper2nd** [comment]
> @kiwibrowser this work hasnt started yet or its already abandoned like any other foss developement by kiwi really shame on you guys just creating new repo to showoffs.

Change to another more up-to-date browser or make your own updated fork. The dev already answered this. I suggest to read his response above your comment.

**Comment by bertandzobrist** [comment]
@bitsper2nd  even  the instructions for forking a new updated one is not clear it will only create vanilla chromium not with extension support at least tell the dev to provide some clear infos at the end of the day its a foss right so one should respect that word and world i guess.

**Comment by kiwibrowser** [comment]
@bertandzobrist It's not true, you can press fork and you get a fully functional Chromium with extensions, there is even all the build scripts directly in GitHub (and you can see them at work because they generate APK)

**Comment by bertandzobrist** [comment]
@kiwibrowser but i tried few months back it only produced vanila chrome w/o extensions anyway i will give a try

**Comment by kiwibrowser** [comment]
Please do, you were right, and now it really changed, I have spent lot of time to change it and actually provide the whole build (and I can assure you, plus you can see they really work end-to-end).

Regarding updates, I am counting on better funded projects to take over some of the heavy work to keep the engine up-to-date.

Once it's done, I'll automatically get the source-code of Kiwi in sync with the latest version, and these scripts will be public.

**Comment by bertandzobrist** [comment]
@kiwibrowser ok man thanks :+1:

**Comment by bertandzobrist** [comment]
OH Browser from Play Store is KIWI KILLER check out it has all the goodies of Kiwi

**Comment by joergRossdeutscher** [comment]
@bertandzobrist Hm? 

- Not OpenSource, some unknown vendor from outside EU regulation (India) to trust on
- Another annoying “design revolution“ like sucking buttons on bottom or gestures
- No permissions for location, camera, microphone, so how can I chat or use maps with it?
- No Addons
- Again a “builtin adblocker“ instead of AddOn of my choice

Feel free to go with it, I personaly see no reason to install, but a dozen to keep away from it.

**Comment by bertandzobrist** [comment]
@joergRossdeutscher Kiwi has just become opensource now i guess you are a very new user of Kiwi

**Comment by kiwibrowser** [comment]
@joergRossdeutscher @bertandzobrist he was answering about the other browser

**Comment by feralghost** [comment]
@kiwibrowser i just built from source following the guide but to my surprise the version is still 77.xxxx.116 where is the latest version of chromium u taking about sorry for the rudeness man it takes a lot of time.

**Comment by kiwibrowser** [comment]
@feralghost You are absolutely right, I'm keeping track of upstreaming efforts and hope to bring good news where Kiwi patches can be ported to the latest version. For now, some of the bigger efforts are done by ungoogled-chromium and brave/brave-core/tree/android_extensions.

The repo correspond to the current version @ Play Store.

I will write whenever I see changes there, and if you see it before me, feel free to ping.

Enjoy the day,

**Comment by Dev-i-l** [comment]
"better funded" - arnaud has recently been working with samsung... nothing suspicious at all...

**Comment by Dev-i-l** [comment]
"better funded" - arnaud has recently been working with samsung... nothing suspicious at all...

**Comment by bitsper2nd** [comment]
> "better funded" - arnaud has recently been working with samsung... nothing suspicious at all...

Yes, there is nothing suspicious about [working together to bring extension support](https://9to5google.com/2020/05/30/kiwi-browser-samsung-extensions-chromium-android/) to the rest on mobile. 😉
